File: connectors/connectors-rabbitmq/src/main/java/com/chaojianok/flink/connectors/rabbitmq/java/sink/RabbitMQSink.java
package com.chaojianok.flink.connectors.rabbitmq.java.sink;

import org.apache.flink.api.common.serialization.SerializationSchema;
import org.apache.flink.streaming.connectors.rabbitmq.RMQSink;
import org.apache.flink.streaming.connectors.rabbitmq.common.RMQConnectionConfig;

import java.io.IOException;
import java.util.Map;

public class RabbitMQSink<IN> extends RMQSink<IN> {
    public RabbitMQSink(RMQConnectionConfig rmqConnectionConfig, String queueName, SerializationSchema schema) {
        super(rmqConnectionConfig, queueName, schema);
    }

    protected void setupQueue() throws IOException {
        if (this.queueName != null) {
            this.channel.queueDeclare(this.queueName, true, false, false, (Map) null);
        }
    }
}

File: connectors/connectors-rabbitmq/src/main/scala/com/chaojianok/flink/connectors/rabbitmq/scala/sink/RabbitMQSink.scala
package com.chaojianok.flink.connectors.rabbitmq.scala.sink

import java.io.IOException

import org.apache.flink.api.common.serialization.SerializationSchema
import org.apache.flink.streaming.connectors.rabbitmq.RMQSink
import org.apache.flink.streaming.connectors.rabbitmq.common.RMQConnectionConfig


class RabbitMQSink[IN](rmqConnectionConfig: RMQConnectionConfig, queueName: String, schema: SerializationSchema[IN]) extends RMQSink[IN](rmqConnectionConfig, queueName, schema) {
  @throws[IOException]
  override protected def setupQueue(): Unit = {
    if (this.queueName != null) this.channel.queueDeclare(this.queueName, true, false, false, null)
  }
}
